assignment/tool/src/java/org/sakaiproject/assignment/entityproviders/AssignmentEntityProvider.java

```diff
@@ -1294,6 +1295,34 @@ public SimpleSubmission(AssignmentSubmission as, boolean anonymousGrading) {
                     this.late = dateSubmitted.compareTo(as.getAssignment().getDueDate()) > 0;
                 }
                 this.submittedAttachments = as.getAttachments();
+
+                SecurityAdvisor securityAdvisor = (String userId, String function, String reference) -> {
+
+                    if (ContentHostingService.AUTH_RESOURCE_READ.equals(function)) {
+                        return SecurityAdvisor.SecurityAdvice.ALLOWED;
+                    } else {
+                        return SecurityAdvisor.SecurityAdvice.NOT_ALLOWED;
+                    }
+                };
+
+                try {
+                    securityService.pushAdvisor(securityAdvisor);
```

Comment:
:( securityAdivisor

Reply:
What do you suggest? Switch to an admin user session? I don't know what to do in these circumstances.

Reply:
That's much more dangerous, especially if on a user thread.

Reply:
I could switch to the owner of the resource, create a temporary session while doing the business.

Reply:
I don't think anyone wants you to jump sessions! I think what ern is saying is that it's too bad we still have the advisor pattern

Reply:
I attach a lot of meaning to all of Earle's comments :) I tried a jumping session approach, but  the idea of a worker pool with loads of threadlocal session data floating around spooked me a bit.

Reply:
Yeah, session swapping is really, really bad. We should never do it. Sometimes our only option is a `SecurityAdvisor`.

Reply:
Sam hit it on the head

Reply:
@bjones the SecurityAdvisor exists simply because the API doesn't accommodate the need to do something without a specific permission.

We all agree there are times when a user actions should do something regardless of the permission they hold. The SecurityAdvisor itself is evidence of this need yet the API doesn't allow us to do it.
If the API had accounted for this then there will be no need for the SecurityAdvisor and the code would be safer because you don't need to elevate someone's permissions.

What I would like people to do is to only use the SecurityAdvisor as a last resort, it's a bad pattern and we should look to limit its use whenever possible.

Reply:
Yeah, that's why I said sometimes it's our only option short of redesigning entire services.